test/src/tests-utils.ts

```diff
@@ -2,18 +2,86 @@ import {
     converted,
     decoded,
     findKeyInObject,
+    generateHash,
+    generateUniqueId,
+    getRampNumber,
     inArray,
     isDataPlanSlug,
     isEmpty,
     isObject,
     isStringOrNumber,
     parseNumber,
     parseStringOrNumber,
-    returnConvertedBoolean
+    returnConvertedBoolean,
 } from '../../src/utils';
 import { expect } from 'chai';
 
 describe('Utils', () => {
+    describe('#generateHash', () => {
+        it('should a hash number with a valid input', () => {
+            expect(generateHash('A'), 'A').to.equal(97);
+            expect(generateHash('false')).to.equal(97196323);
+            expect(generateHash('3569038')).to.equal(-412991536);
+            expect(generateHash('TestHash')).to.equal(-1146196832);
+            expect(generateHash('mParticle'), 'mParticle String').to.equal(
+                1744810483
+            );
+            expect(
+                generateHash('d71b49a6-4248-4581-afff-abb28dada53d')
+            ).to.equal(635757846);
+        });
+
+        it('returns 0 when hashing undefined or null', () => {
+            expect(generateHash(null)).to.equal(0);
+            expect(generateHash(undefined)).to.equal(0);
+
+            // Use bad values to verify expected fail cases
+            expect(typeof generateHash(false as unknown as string)).to.equal(
+                'number'
+            );
+            expect(generateHash(false as unknown as string)).to.not.equal(0);
+        });
+    });
+
+    describe('#generateUniqueId', () => {
+        it('generate a random value', () => {
+            expect(generateUniqueId()).to.be.ok;
+            expect(generateUniqueId().length).to.equal(36);
+            expect(typeof generateUniqueId()).to.equal('string');
+
+            // Tests format to be broken up by 5 hypens
+            expect(generateUniqueId().split('-').length).to.equal(5);
+
+            window.crypto.getRandomValues = undefined;
+            expect(generateUniqueId()).to.be.ok;
+            // old browsers may return undefined despite
+            // defining the getRandomValues API.
```

Comment:
I'm not sure what this means.  Can you clarify?

Reply:
I'm merely moving an existing function and comment from one module to another: https://github.com/mParticle/mparticle-web-sdk/commit/9ad0b7c3deb73dfc5f40258c4cdd4ebffaeb92e8#diff-2aa8a2156523a4dec34b9e1e21a020a598189186c8fc47b26958facc11dbb9f9R196

Should we ask the original contributor? 😉